lib/WorseReflection/Bridge/TolerantParser/Diagnostics/UndefinedVariableProvider.php

```diff
@@ -25,13 +25,19 @@ public function enter(NodeContextResolver $resolver, Frame $frame, Node $node):
         if ($node->parent?->parent instanceof PropertyDeclaration) {
             return [];
         }
-        foreach ($frame->locals()->byName($node->getName()) as $variable) {
-            if ($variable->wasDefinition()) {
-                return [];
+
+        $name = $node->getName();
+
+        if ($name) {
+            foreach ($frame->locals()->byName($name) as $variable) {
+                if ($variable->wasDefinition()) {
+                    return [];
+                }
             }
+        } else {
+            $name = 'UNDEFINED';
         }
```

Comment:
in general avoid `else`  - but can we not simply `if (!$name) { return; }` ? why set it to `UNDEFINED`?

Reply:
Because I did not look at the code properly and did not realise it :D